web/src/features/hfiCalculator/components/stationAdmin/newStationForm.test.tsx

```diff
@@ -1,94 +1,143 @@
 import { render } from '@testing-library/react'
-import { AdminStation } from 'features/hfiCalculator/components/stationAdmin/AddStationModal'
+import { FuelType } from 'api/hfiCalculatorAPI'
+import {
+  AdminStation,
+  BasicWFWXStation
+} from 'features/hfiCalculator/components/stationAdmin/AddStationModal'
 import NewStationForm from 'features/hfiCalculator/components/stationAdmin/NewStationForm'
 import React from 'react'
 
 describe('NewStationForm', () => {
-  const renderNewStationForm = (newStation: AdminStation, invalid: boolean) => {
-    const { getByTestId, queryByText } = render(
-      <NewStationForm newStation={newStation} invalid={invalid} />
+  const setNewStationMock = jest.fn()
+  const setInvalidMock = jest.fn()
+
+  beforeEach(() => {
+    setNewStationMock.mockReset()
+    setInvalidMock.mockReset()
+  })
+
+  const renderNewStationForm = (
+    newStation: AdminStation,
+    invalid: boolean,
+    stationAddedError: string | null
+  ) => {
+    return render(
+      <NewStationForm
+        newStation={newStation}
+        invalid={invalid}
+        setNewStation={setNewStationMock}
+        setInvalid={setInvalidMock}
+        stationAddedError={stationAddedError}
+      />
     )
-    return { getByTestId, queryByText }
   }
   describe('valid states', () => {
     it('should not render error message when new station has not been edited', () => {
-      const { queryByText } = renderNewStationForm({ dirty: false }, false)
+      const { queryByText } = renderNewStationForm({ dirty: false }, false, null)
 
       const errorMessage = queryByText('Please complete empty fields to continue')
       expect(errorMessage).not.toBeInTheDocument()
```

Comment:
Seems a bit risky to be testing again a search text message - this test will always pass if we happen to later change the error message text on the frontend.
But also, I can't think of a better solution right now...

Reply:
Not true, if you change the text this test will fail.